index.js

```diff
@@ -115,11 +120,19 @@ class ServerlessPlugin {
           this.serverless.service.provider.environment[key] = envVars[key]
         })
       } else {
-        if (this.logging) {
+        const errorMsg = 'DOTENV: Could not find .env file.';
+
+        if(this.fileRequired === true) {
+          throw Object.assign(new Error(errorMsg), { type: errorTypes.FILE_NOT_FOUND });
+        } else if (this.logging) {
           this.serverless.cli.log('DOTENV: Could not find .env file.')
         }
       }
     } catch (e) {
+      if(this.fileRequired && e.type === errorTypes.FILE_NOT_FOUND) {
```

Comment:
```suggestion
      if(e.type === errorTypes.HALT) {
```
or something similar. No need to be specific, just specify "halt" or "don't halt".

Reply:
"halt" is the right word! :-D